pkg/server/upload.go

```diff
@@ -243,25 +243,13 @@ func validateKeys(ctx context.Context, w http.ResponseWriter, keys []*pb.Tempora
 	max := ints[len(ints)-1]
 	maxEnd := max + 144
 
-	if maxEnd-min > (144 * 14) {
+	if maxEnd-min > (144 * 15) {
 		requestError(
-			ctx, w, nil, "sequence of rollingStartIntervalNumbers exceeds 14 days",
+			ctx, w, nil, "sequence of rollingStartIntervalNumbers exceeds 15 days",
 			http.StatusBadRequest, uploadError(pb.EncryptedUploadResponse_INVALID_ROLLING_START_INTERVAL_NUMBER),
 		)
 		return false
 	}
 
-	lastEnd := 0
-	for _, rsn := range ints {
-		if rsn < lastEnd {
-			requestError(
-				ctx, w, nil, "overlapping or duplicate rollingStartIntervalNumbers",
```

Comment:
We will need to add some nuanced checks back in I assume?

Reply:
Yes once we know more about the spec

Reply:
But overlapping or duplicate RSIN are explicitly allowed